mm_signals.py

```diff
@@ -0,0 +1,258 @@
+        trdEx=deal["trdEx"]
+    )
+
+
+def make_orders(data: dict) -> list:
+    if "depth" in data:
+        depth = data["depth"]
+        ask_list = depth["ntvAggAskList"]
+        bid_list = depth["ntvAggBidList"]
+        orders = list()
+
+        for ask in ask_list:
+            for ask_order in ask["orderMap"]:
+                orders.append(
+                    Order("ASK", ask["price"],
+                          ask_order["volume"], ask_order["marketName"]))
+        for bid in bid_list:
+            for bid_order in bid["orderMap"]:
+                orders.append(
+                    Order("BID", bid["price"],
+                          bid_order["volume"], bid_order["marketName"]))
+
+        return orders
+
+    if "deal" in data:
+        deal: dict = data["deal"]
+
+        deal_colors = {
+            "S": "[red]",
+            "B": "[green]",
+            "G": "[white on dark green]",
+            "N": "[grey62]",
+            "L": "[white on red3]"
+        }
+
+        color_str = deal_colors[deal["trdBs"]]
+        base_str = f"{color_str}{deal['tradeTime']} on {deal['trdEx']} "
+
+        if deal["volume"] == "100":
+            console.print(f"{base_str} I need Shares.")
+        if deal["volume"] == "200":
+            console.print(
+                f"{base_str} I need Shares badly! Do not take stock down.")
+        if deal["volume"] == "300":
+            console.print(
+                f"{base_str} Take/Taking the stock down so I can load shares.")
+            create_deal(deal)
+        if deal["volume"] == "400":
+            console.print(f"{base_str} Keep trading it sideways.")
+            create_deal(deal)
+        if deal["volume"] == "500":
+            direction = "up ⬆️" if deal["trdBs"] in ("B", "G") else "down ⬇️"
+            console.print(f"{base_str} Gap the stock {direction} ")
+            create_deal(deal)
+
+        if deal["volume"] == "505":
+            console.print(f"{base_str} I am short on shares.")
+            create_deal(deal)
+
+        if deal["volume"] == "600":
+            console.print(f"{base_str} Apply resistance at {deal['price']}")
+            create_deal(deal)
+
+        if deal["volume"] in ("700", "777"):
+            console.print(f"{base_str} Move the price up.")
+            create_deal(deal)
+
+        if deal["volume"] == "800":
+            console.print(f"{base_str} Volume coming.")
+            create_deal(deal)
+
+        if deal["volume"] == "900":
+            console.print(f"{base_str} Let it trade freely.")
+            create_deal(deal)
+
+        if deal["volume"] == "911":
+            console.print(f"{base_str} NEWS PENDING.")
+            create_deal(deal)
+
+        if deal["volume"] == "1000":
+            console.print(f"{base_str} Don't let it run!")
+            create_deal(deal)
+
+        if deal["volume"] == "2100":
+            console.print(f"{base_str} Let it run!")
+            create_deal(deal)
```

Comment:
I rolled this up in this manner, would you like to use this instead?

```py
        volumes = {
            "100": { "msg": "I need Shares", "create_deal": False },
            "200": { "msg": "I need Shares badly do not take stock down", "create_deal": False },
            "300": { "msg": "Take/Taking the stock down so I can load shares", "create_deal": True },
            "400": { "msg": "Keep trading it sideways", "create_deal": True },
            "500": { "msg": "Gap the stock ", "create_deal": True, "direction": { "B": "up ⬆️", "G": "up ⬆️", "S": "down ⬇️", "L": "down ⬇️" } },
            "505": { "msg": "I am short on shares", "create_deal": True },
            "600": { "msg": "Apply resistance at ", "create_deal": True, "suffix": "price" },
            "700": { "msg": "Move the price up", "create_deal": True },
            "777": { "msg": "Move the price up", "create_deal": True },
            "800": { "msg": "Volume coming", "create_deal": True },
            "900": { "msg": "Let it trade freely", "create_deal": True },
            "911": { "msg": "NEWS PENDING", "create_deal": True },
            "1000": { "msg": "Don't let it run!", "create_deal": True },
            "2100": { "msg": "Let it run!", "create_deal": True },
        }

        volume = volumes.get(deal['volume'])

        msg = volume.get('msg')
        direction = volume.get('direction', {}).get(deal['trdBs'], "")
        suffix = deal.get(volume.get("suffix"), "")

        console.print(f"{base_str}{msg}{direction}{suffix}")
        if volume.get('create_deal'):
            create_deal(deal)
```

If not, I may open a separate PR once this is merged.

Reply:
Thank you for helping review!
This looks really cool and easy to change, but it took me while to understand how it worked so i feel readability went down so it is a trade off, but I like it. Please do make your own pr after this is merged!

Reply:
Sounds good, I was worried about readability too with the special handling for `500` and `600`. I'll give it another run through for readability and see what I can do!

Reply:
Sounds good! I will set these as resolved, and I will wait for your PR!